src/python/pants/engine/target.py

```diff
@@ -691,79 +683,73 @@ def types(self) -> Tuple[Type[Target], ...]:
 
 
 # -----------------------------------------------------------------------------------------------
-# Generated subtargets
+# Target generation
 # -----------------------------------------------------------------------------------------------
 
+_Tgt = TypeVar("_Tgt", bound=Target)
 
-def generate_subtarget_address(build_target_address: Address, *, full_file_name: str) -> Address:
-    """Return the address for a new target based on the BUILD target, but with a more precise
-    `sources` field.
 
-    The address's target name will be the relativized file, such as `:app.json`, or `:subdir/f.txt`.
+@union
+@dataclass(frozen=True)
+class GenerateTargetsRequest(Generic[_Tgt]):
+    target_class: ClassVar[type[_Tgt]]
```

Comment:
I had to read the applicable `@rule` to figure out that `target_class` was the generator's target type. (Since the existing usages are all essentially identity mappings, it is ambiguous to a reader with the current naming.)

 Maybe call it `generate_for` similar to how `InferDependenciesRequest` uses `infer_from` and `InjectDependenciesRequest` uses `inject_for`?

Also a comment covering these fields would be useful for readers.

Reply:
Good idea. What about generate_from?

Reply:
Works for me.